### ISSUE TITLE
Trigger thinking animation on TTS End event

### ISSUE THREAD

**Issue by elggem** [hansonrobotics/hr_reflexes#3]
Trigger randomly from a set of response animations, like look top-right/left or nodding. This provides immediate feedback ala Alexa's LED display and make the robot appear more responsive.

  - [ ] @wenwei-dev: the topic that tells you tts ends is called speech_events. If tts ends, it will send the string message "speechend".
  - [ ] Define a set of thinking animations
  - [ ] Trigger them with some variability.

**Comment by jmossbridge** [comment]
With time, I think it would be cool to specifically trigger certain animations rather than randomly. Like looking  straight at the person after they have  spoken a long time, then  after a delay, looking away  to think about  it. The delay  and  the looking away could be timed to be a specific proportion of the duration  of what they said (like 1/10th the duration)...the look away should be to Sophia's left  (our right) when she is "considering" something from the past or making a  judgment,  should be to  her right (our left)  when she is considering something new  or about the future or being supportive  (so for loving AI, most looks should be to her right  (our  left) after looking in the eyes).

**Comment by jmossbridge** [comment]
Also, when "thinking" hard, people contract the muscles between their eyes. Worth adding this animation (and by  the way, this should be in mirroring -- is it?)

**Comment by elggem** [comment]
Absolutely the thinking animation should be context dependent. This should be specified in the Ghost rulebase, once the time has come! For now I will trigger randomly with universally acceptable animations, and later we might fake context based on some cues you mentioned, like how long the last uttering was, etc. 

No, mirroring only includes moving eyebrows down (and quite harshly). I will look into it, but I doubt Realsense gives this data...

**Comment by elggem** [comment]
Idea for detecting speechend event:

´´´
rostopic echo /sophia6/audio_sensors
---
Decibel: 37.6373939514
Frequency: 625.374389648
SuddenChange: False
Speech: False
---
´´´

Calculate moving window and detect big drops, trigger animation if that happens.

**Comment by jmossbridge** [comment]
@elggem --  my concern  is the  microphone is turned off once google voice detects the end of speech. So is there a second source for the dB reading?

**Comment by elggem** [comment]
Yes, yes this is independent. Actually the mic isnt turned off, just the STT detection.
